### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by vlasec**
Initial implementation of the challenge. It doesn't contain any tests because of time constraints. Resolving HTTP redirects took quite a bit of effort that could have gone into the tests if the requirements weren't that it works with two websites that both use redirects.

Some further optimization might be advised, but the application works as intended.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import Dependencies._
+
+ThisBuild / version := "0.1.0-SNAPSHOT"
+
+ThisBuild / scalaVersion := "2.12.17"
```
Comment (btomala) [inline build.sbt:5]: Why you are using such an old version of scala?

File: src/main/scala/cz/vlasec/gwi/csvimport/CirceSupport.scala [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import scala.concurrent.Future
+
+// from https://gist.github.com/mattroberts297/c531a4e9e525d6a18cbf8889ab5f3dec
+/**
+ * To use circe for json marshalling and unmarshalling:
+ *
+ * import CirceSupport._
+ * import io.circe.generic.auto._
+ */
+object CirceSupport {
```
Comment (btomala) [inline src/main/scala/cz/vlasec/gwi/csvimport/CirceSupport.scala:21]: You can use https://github.com/hseeberger/akka-http-json instead. However, I just read that it's no longer maintained.

File: src/main/scala/cz/vlasec/gwi/csvimport/HttpSourceror.scala [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    }
+  }
+
+  private def busy(url: String, replyTo: SourceConsumerRef, redirects: Int = 0, clone: Boolean = false)
+  : Behavior[HttpSourcerorCommand] = Behaviors.setup { context =>
+    implicit val classicSystem: ActorSystem = context.system.classicSystem
+    implicit val executionContext: ExecutionContext = context.system.executionContext
+
+    Http().singleRequest(Get(url))
+      .onComplete(context.self ! HandleResponse(_))
```
Comment (btomala) [inline src/main/scala/cz/vlasec/gwi/csvimport/HttpSourceror.scala:50]: What will happen if CSV is too big to fit into application memory?

File: src/main/scala/cz/vlasec/gwi/csvimport/Routes.scala [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  import io.circe.generic.auto._
+
+  private def task(taskServiceRef: ActorRef[ServiceCommand])(implicit scheduler: Scheduler): Route = {
+    implicit val timeout: Timeout = 200.millis
+    pathPrefix("task") {
+      concat(
+        pathEnd {
+          concat(
+            get {
+              val response: Seq[TaskStatusReport] = Await.result(
```
Comment (btomala) [inline src/main/scala/cz/vlasec/gwi/csvimport/Routes.scala:29]: Why do you need Await here? Why not pass the future to the complete function? Also if you use JsonMarshallers you don't need to pass ContentType explicitly.

**Review by btomala — COMMENTED** [review]
Thanks for writing the code. The code is nice! I left a few minor comments. I don't like that Awiats are everywhere, don't see the need for them. 

Unfortunately, as you already hear we stopped recruiting for this position, for now. Thanks again for your effort.

File: build.sbt (hunk 1 above)
Comment (vlasec) [reply build.sbt:5]: I've been using Scala 2.12 in my work so far, so I found it convenient to stay in the same version for now.

I know newer Scala versions have a lot of improvements, and I'll gladly learn newer Scala when I have some use for it.

File: src/main/scala/cz/vlasec/gwi/csvimport/HttpSourceror.scala (hunk 3 above)
Comment (vlasec) [reply src/main/scala/cz/vlasec/gwi/csvimport/HttpSourceror.scala:50]: As far as I know, this response should only hold a stream of data. I don't know the underlying implementation though, so there could be nasty surprises.

For example my former teammate found out that Netty always copies the response into a single array of bytes at one point, resulting in NegativeArraySizeException. One wouldn't expect such a folly in a widely used library.

File: src/main/scala/cz/vlasec/gwi/csvimport/Routes.scala (hunk 4 above)
Comment (vlasec) [reply src/main/scala/cz/vlasec/gwi/csvimport/Routes.scala:29]: Thank you, this is a good point. To be honest, I still have a lot to learn about Akka HTTP, and Akka in general.

**Pull request closed without merging**